### PR TITLE
Fix resource detection

### LINKED ISSUE #25: Fix resource detection, missing labels on deployed resources.
**Is this a BUG REPORT or FEATURE REQUEST?**:
BUG
**What happened**:
Deployed resources in workflow template should be labeled with application labels. 
The labels are missing which are required for resource detection.
**What you expected to happen**:
Add labels to Deployments
**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Addon Manager version
- Kubernetes version :
```
$ kubectl version -o yaml
```

**Other debugging information (if applicable)**:
- Addon status:
```
$ kubectl describe addon <addon-name>
```
- controller logs:
```
$ kubectl logs <addon-manager-pod>
```

### PR DESCRIPTION
**Pull request by kevdowney**
Fixes #25 

Adds labels to deployed resources in workflow templates.

### CODE REVIEW

File: config/crd/bases/addonmgr.keikoproj.io_addons.yaml [hunk 1]
```diff
@@ -665,6 +665,9 @@ spec:
                     type: string
                 type: object
               type: array
+            starttime:
+              format: int64
```
Comment (pzou1974) [inline config/crd/bases/addonmgr.keikoproj.io_addons.yaml]: starttime is  epoch time in second?

File: pkg/workflows/workflow.go [hunk 2]
```diff
@@ -301,149 +301,142 @@ func (w *workflowLifecycle) configureWorkflowArtifacts(wf *unstructured.Unstruct
...
-		var newData []string
-		for _, obj := range resources {
-			resource := make(map[string]interface{})
-			err := yaml.Unmarshal([]byte(obj), &resource)
+	if foundArtifacts {
+		for _, artifact := range artifacts.([]interface{}) {
+			artifact := artifact.(map[string]interface{})
+			data, _, err := unstructured.NestedString(artifact, "raw", "data")
 			if err != nil {
-				return fmt.Errorf("unable to unmarshall artifact: %s", obj)
```
Comment (pzou1974) [inline pkg/workflows/workflow.go]: you removed the logging.  make sure you can catch the exact error at which step laster.

File: config/crd/bases/addonmgr.keikoproj.io_addons.yaml (hunk 1 above)
Comment (kevdowney) [reply config/crd/bases/addonmgr.keikoproj.io_addons.yaml]: This is carry-over from your status change, this is auto-generated.

File: pkg/workflows/workflow.go (hunk 2 above)
Comment (kevdowney) [reply pkg/workflows/workflow.go]: This is not log, just error but now will bubble up the underlying error as it was obscured before.

**Pull request merged**